scvi/model/base/_base_model.py

```diff
@@ -92,13 +112,55 @@ def to_device(self, device: Union[str, int]):
     def device(self):
         return self.module.device
 
+    @staticmethod
+    def _get_setup_method_args(**setup_locals) -> dict:
+        """
+        Returns a dictionary organizing the arguments used to call ``setup_anndata``.
+
+        Must be called with ``**locals()`` at the start of the ``setup_anndata`` method
+        to avoid the inclusion of any extraneous variables.
+        """
+        setup_locals.pop("adata")
+        cls = setup_locals.pop("cls")
+        model_name = cls.__name__
+        setup_kwargs = dict()
+        for k, v in setup_locals.items():
+            if k not in _SETUP_INPUTS_EXCLUDED_PARAMS:
+                setup_kwargs[k] = v
+        return {_MODEL_NAME_KEY: model_name, _SETUP_KWARGS_KEY: setup_kwargs}
+
     @classmethod
-    def get_anndata_manager(
+    def register_manager(cls, adata_manager: AnnDataManager):
+        """
+        Registers an :class:`~scvi.data.anndata.AnnDataManager` instance with this model class.
+        """
+        adata_id = adata_manager.get_adata_uuid()
+        manager_store_key = _ManagerStoreKey(adata_id=adata_id, model_instance_id=None)
+        cls._manager_store[manager_store_key] = adata_manager
+
+    def _register_manager_for_instance(self, adata_manager: AnnDataManager):
+        """
+        Registers an :class:`~scvi.data.anndata.AnnDataManager` instance with this model instance.
+
+        Creates a model-instance specific mapping in ``cls._manager_store`` for this
+        :class:`~scvi.data.anndata.AnnDataManager` instance.
+        """
+        adata_id = adata_manager.get_adata_uuid()
+        manager_store_key = _ManagerStoreKey(
+            adata_id=adata_id, model_instance_id=self.id
+        )
+        self._manager_store[manager_store_key] = adata_manager
```

Comment:
here you're using self, and this key will be different than in the `register_manager` call on 139. Is that intended? Shouldn't we overwrite?

Reply:
I got it as I continued the review, there has to be a better way we can communicate what's going on. Maybe we have two stores so the code is more readable?